stan/math/prim/scal/fun/rising_factorial.hpp

```diff
@@ -51,11 +55,13 @@ namespace stan {
      */
     template<typename T1, typename T2>
     inline typename boost::math::tools::promote_args<T1, T2>::type
-    rising_factorial(const T1 x, const T2 n) {
-      using std::exp;
-      return exp(log_rising_factorial(x, n));
+    rising_factorial(const T1& x, const T2& n) {
+      if (is_nan(x) || is_nan(n))
+        return std::numeric_limits<double>::quiet_NaN();
```

Comment:
Do we really want this to return NaN rather than throw?  All of our other functions throw, so I think this one should, too.  There's no worry about backward compatibility here as none of this error behavior is actually documented  properly in the Stan manual (that's a bug, not a feature of the manual).